inc/Ui/MediaDiff.php

```diff
@@ -127,6 +130,11 @@ public function show()
 
         if ($auth < AUTH_READ || !$this->id || !$conf['mediarevisions']) return;
 
+        // detect ajax request  FIXME could eliminate $this->preference['fromAjax'] ?
+        $isAjaxRequest = (isset($_SERVER['HTTP_X_REQUESTED_WITH'])
+            && strtolower($_SERVER['HTTP_X_REQUESTED_WITH']) == 'xmlhttprequest'
+        );
+
```

Comment:
I prefer a setter instead of this magic checking somewhere internally.

Reply:
ok. I am no clear why `$_SERVER['HTTP_X_REQUESTED_WITH` not used, likely some reason may exists. I will remove the code.

Reply:
The Ajax call is going via inc/Ajax.php, so in my understanding understanding this kind of code should live there.. I guess that the Ajax mode needs another kind of output.

hmm, with a bit more looking around I guess now that this preference is only set for mediadiff. And that mediadiff always arrives via inc/Ajax.php (Media Manager is doing its tasks via Ajax). So maybe this setting is superfluous?